fml/mapping.cc

```diff
@@ -102,6 +101,47 @@ const uint8_t* NonOwnedMapping::GetMapping() const {
   return data_;
 }
 
+// MallocMapping
+MallocMapping::MallocMapping() : data_(nullptr), size_(0) {}
+
+MallocMapping::MallocMapping(const uint8_t* data, size_t size)
+    : data_(data), size_(size) {}
+
+MallocMapping::MallocMapping(fml::MallocMapping&& mapping)
+    : data_(mapping.data_), size_(mapping.size_) {
+  mapping.data_ = nullptr;
+  mapping.size_ = 0;
+}
+
+MallocMapping::~MallocMapping() {
+  if (data_) {
+    free(const_cast<uint8_t*>(data_));
+    data_ = nullptr;
+  }
+}
+
+MallocMapping MallocMapping::Copy(const void* begin, size_t length) {
+  auto result =
```

Comment:
Error handling for OOM cases on malloc failures.

Reply:
done